### DISCUSSION TITLE
Issues in LDT generated lis-input file

### DISCUSSION THREAD

**Discussion by Arijit22IITDelhi** [NASA-LIS/LISF#1530] [Q&A] [answered]
Hello, I was trying to generate lis input file using ldt.parameter file with the configuration as follows:

#Overall driver options
LDT running mode:                       "LSM parameter processing"
Processed LSM parameter filename:       ./ldt_param_noahmp401_SA.nc
LIS number of nests:                    1
Number of surface model types:          1
Surface model types:                    "LSM"
Land surface model:                     "Noah-MP.4.0.1"
Lake model:                             none
Water fraction cutoff value:            0.5

Number of met forcing sources:          0
Met forcing sources:                    "MERRA2"
Met spatial transform methods:          "neighbor"  # bilinear
Topographic correction method (met forcing):  "lapse-rate"
Temporal interpolation method (met forcing): linear
LDT diagnostic file:                    ldtlog
LDT output directory:                   OUTPUT
Undefined value:                       -9999.0

#Processor layout
Number of ensembles per tile:           1
Number of processors along x:           1
Number of processors along y:           1

#LIS domain
Map projection of the LIS domain:      latlon
Run domain lower left lat:             5.05 
Run domain lower left lon:             64.55
Run domain upper right lat:            37.95
Run domain upper right lon:            99.95 
Run domain resolution (dx):            0.1
Run domain resolution (dy):            0.1

# Parameters
Landcover data source:        MODIS_Native 
Landcover classification:       IGBPNCEP 
Landcover file:            ./INPUT/LS_PARAMETERS/noah_2dparms/igbp.bin
Landcover spatial transform:     tile
Landcover fill option:           none # Set to none if creating land mask
Landcover map projection:       latlon

Create or readin landmask:    create
Landmask data source:         MODIS_Native
Landmask spatial transform:   none

# Crop-Irrigation parameters
Incorporate crop information:     .false.
Irrigation type data source:      none
Irrigation fraction data source:  none

# Soil Parameters
Soil fraction data source:       none
Soils spatial transform:         none
Soils fill option:               none
Soils map projection:            latlon

Soil texture data source:       STATSGOFAO_Native
Soil texture map:               ./INPUT/LS_PARAMETERS/noah_2dparms/topsoil30snew
Soil texture spatial transform:   mode
Soil texture fill option:         neighbor
Soil texture fill value:          6
Soil texture fill value for Antarctica:   16
Soil texture fill radius:         5
Soil texture map projection:     latlon
Soil texture force exclusion of water points during fill: true

# Topography parameters -- Elevation, Slope, Aspect
Elevation data source:        SRTM_Native
Elevation number of bands:        1
Elevation map:       	./INPUT/LS_PARAMETERS/topo_parms/SRTM30/raw_wgtopo30antarc
Elevation fill option:           none # Preserve elevation for water bodies
Elevation fill value:             0
Elevation fill radius:            5

Slope data source:        SRTM_Native
Slope number of bands:        1
Slope map:     		./INPUT/LS_PARAMETERS/topo_parms/SRTM30/raw_wgtopo30antarc
Slope fill option:            none # Preserve slope for water bodies
Slope fill value:             0
Slope fill radius:            5.

Aspect data source:        SRTM_Native
Aspect number of bands:        1
Aspect map:       	./INPUTLS_PARAMETERS/topo_parms/SRTM30/raw_wgtopo30antarc
Aspect fill option:            none # Preserve aspect ratio for water bodies
Aspect fill value:             3.14159
Aspect fill radius:            5.

Topography spatial transform:     average 
Topography map projection:       latlon

# LSM-specific parameters
Albedo data source:          NCEP_Native
Albedo map:                 ./INPUT/LS_PARAMETERS/noah_2dparms/albedo 
Albedo climatology interval:    monthly
Albedo spatial transform:       budget-bilinear
Albedo fill option:             neighbor
Albedo fill value:              0.15
Albedo fill radius:             5
Albedo map projection:          latlon

Max snow albedo data source: Barlage_Native
Max snow albedo map:        ./INPUT/LS_PARAMETERS/noah_2dparms/maximum_snow_albedo.hdf
Max snow albedo spatial transform:     average
Max snow albedo fill option:           neighbor
Max snow albedo fill value:           0.3
Max snow albedo fill radius:          5
Max snow albedo map projection:        latlon

Greenness data source:         NCEP_Native
Greenness fraction map:        ./INPUT/LS_PARAMETERS/noah_2dparms/gfrac
Greenness climatology interval:    monthly
Calculate min-max greenness fraction:  .false.
Greenness maximum map:         ./INPUT/LS_PARAMETERS/noah_2dparms/gfrac_max.asc
Greenness minimum map:         ./INPUT/LS_PARAMETERS/noah_2dparms/gfrac_min.asc
Greenness spatial transform:       budget-bilinear
Greenness fill option:             neighbor
Greenness fill radius:             5
Greenness fill value:              0.3
Greenness maximum fill value:     1.0
Greenness minimum fill value:     0.0
Greenness map projection:         latlon

#Slope type inputs  
Slope type data source:         CONSTANT 
Slope type map:                 none
Slope type spatial transform:   neighbor                # none | neighbor | mode
Slope type fill option:         neighbor                # none | neighbor
Slope type fill radius:           0                     # Number of pixels to search for neighbor
Slope type fill value:            1.                    # Recommended static value to fill where missing
Slope type map projection:      latlon

Bottom temperature data source:   ISLSCP1
Bottom temperature map:          ./INPUT/LS_PARAMETERS/noah_2dparms/SOILTEMP.60
Bottom temperature spatial transform:    budget-bilinear
Bottom temperature fill option:      average      
Bottom temperature fill value:       287.0       
Bottom temperature fill radius:       5.
Bottom temperature topographic downscaling:  lapse-rate
Bottom temperature map projection:      latlon
Noah-MP PBL Height Value: 900.

# ------------------------

After running this using LDT, the resulting output file  "ldt_param_noahmp401_SA.nc", is showing DOMAINMASK as below: 
![DomainMask_NoahMP401](https://github.com/NASA-LIS/LISF/assets/137404009/cb83320b-7b3b-434e-813e-b87d945acc16)
 As we can see, there are some gaps within the domain mask, same goes for other input parameters. Is there any way of smoothening this or anything I need to specify within the LDT config file as above?

**Comment by dmocko** [comment, marked as answer]
I am not sure exactly what you mean by gaps.  Do you mean the inland water points (lakes, rivers, etc.)?

Why do you need those points to be land points, instead of water?

**Reply by Arijit22IITDelhi** [reply]
@dmocko Are they exactly considering almost all inland water points? I mean, if those violet patches within the land area are really signifying water bodies, then there should be more patches downwards near the Gangetic plane!! or is it like they are considering reservoirs only, not the entire stream?
